src/sys/pthread.rs

```diff
@@ -43,17 +43,24 @@ pub fn pthread_kill<T: Into<Option<Signal>>>(thread: Pthread, signal: T) -> Resu
 ///
 /// On linux the name cannot exceed 16 length including null terminator
 #[cfg(all(target_os = "linux", not(target_env = "musl")))]
-pub fn pthread_getname_np(thread: Pthread) -> Result<String> {
-    let mut name = [0u8; 16];
+pub fn pthread_getname_np(thread: Pthread) -> Result<CString> {
+    let mut name: Vec<u8> = Vec::with_capacity(16);
+    unsafe { name.set_len(16); }
```

Comment:
This is technically UB.  You shouldn't call `set_len` until after the the FFI function returns successfully, and you should only set it to as many bytes as the FFI function actually wrote.  In this case, `pthread_getname_np` doesn't actually return the length of the name, so you'll have to fully initialize the buffer anyway.